HW0/P6/P6.txt

```diff
@@ -1 +1,37 @@
 # Your answers here
+P6A
+One result is:
+Hi Job 1
+Hi Job 2
+Hi Job 0
+Hi Job 3
+Bye Job 1
+Bye Job 0
+Bye Job 2
+Hi Job 4
+Hi Job 5
+Hi Job 6
+Bye Job 3
+Hi Job 7
+Bye Job 4
+Bye Job 5
+Bye Job 6
+Hi Job 8
+Bye Job 7
+Hi Job 9
+Bye Job 8
+Bye Job 9
+
+We can see there are unexpected results of the job orders:
+1) The order jobs begin does not match the order we send in code.
+2) The order jobs end does not match the order they begin.
+3) Run this program several times, the sequences may different every time.
+
+1) would be a concern when we need to serve many identical tasks with different owners. In general, we should follow the rule: first come, first served. However, this program cannot guarantee this fiarness.
```

Comment:
I'm not sure I fully understand your concern #1. Is the only problem that the order of allocation is unfair, or do you mean that this can also create problems with the running/output of the jobs?